src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -791,8 +789,33 @@ private async Task<IEnumerable<DiagnosticApiResponse>> ListDetectorsInternal(Run
             if (context.ClientIsInternal)
             {
                 dataProvidersMetadata = GetDataProvidersMetadata(dataProviders);
-            }       
-            var invoker = this._invokerCache.GetEntityInvoker<TResource>(detectorId, context);
+            }
+            EntityInvoker invoker = null;
+            if(tableCacheService.IsStorageAsSourceEnabled())
+            {
+                 invoker = this._invokerCache.GetEntityInvoker<TResource>(detectorId, context);
+                var allDetectors = await this.tableCacheService.GetEntityListByType(context, "Detector");
```

Comment:
QQ - This code is hitting the memcache and not making outbound storage calls every single time, correct?

Reply:
Yup correct. It will first hit memory cache and only if its not present in memory cache we fetch from Storage.